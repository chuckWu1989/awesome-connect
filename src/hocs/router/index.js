import React from 'react';

const router = (method = {}) => (
  WrappedComponent => (
    props => <WrappedComponent {...props} {...method} />
  )
);

export default router;
