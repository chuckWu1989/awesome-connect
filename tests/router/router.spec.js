import React from 'react';
import toJson from 'enzyme-to-json';
import router from '../../src/hocs/router';

const { shallow } = global;

describe('router', () => {
  const WrappedComponent = () => <div />;
  const props = { test: 'test2' };
  it('should render properly', () => {
    const method = { default: 'test' };
    const RouterComponent = router(method)(WrappedComponent);
    const tree = toJson(shallow(<RouterComponent {...props} />));
    expect(tree).toMatchSnapshot();
  });
  it('should render properly if method is not defined', () => {
    const RouterComponent = router()(WrappedComponent);
    const tree = toJson(shallow(<RouterComponent {...props} />));
    expect(tree).toMatchSnapshot();
  });
});
