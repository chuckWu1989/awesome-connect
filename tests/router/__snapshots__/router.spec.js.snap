// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`router should render properly 1`] = `
<WrappedComponent
  default="test"
  test="test2"
/>
`;

exports[`router should render properly if method is not defined 1`] = `
<WrappedComponent
  test="test2"
/>
`;
